eagleEvents/seating_chart_ga.py

```diff
@@ -7,182 +7,346 @@
+    toolbox.register("population_guess", init_population, list, creator.Individual, seed)
+
+    pop = toolbox.population_guess(n=NIND, pct_heuristic=INIT_PCT_GUESS)
+
+    # Evaluate the first generation
+    update_fitnesses(lookup_and_size, pop)
+    if collect_stats:
+        update_stats(mstats, logbook, 0, pop)
+
+    generation_number = 1
+    while not should_terminate(hall_of_fame, generation_number, total_like_preferences, NGEN):
+        pop[:] = do_generation(hall_of_fame, lookup_and_size, pop, NIND)
+        if collect_stats:
+            update_stats(mstats, logbook, generation_number, pop)
+        pop.extend(hall_of_fame)
+        hall_of_fame = sorted(pop, key=lambda indiv: indiv.fitness.values[1] - (indiv.fitness.values[0] * 2),
+                              reverse=True)[:HALL_OF_FAME_SIZE]
+        if log_output:
+            print("Generation: {}".format(generation_number))
+            print("Current Best: {a} dislikes, {b} likes".format(a=hall_of_fame[0].fitness.values[0],
+                                                                 b=hall_of_fame[0].fitness.values[1]))
+        generation_number += 1
+
+    winner = hall_of_fame[0]
+    tables = []
+    for t in range(num_tables):
+        table = Table(event)
+        table.number = t + 1
+        guest_numbers_at_table = winner[t * event.table_size.size: (t + 1) * event.table_size.size]
+        table.guests = list(filter(lambda g: g.number in guest_numbers_at_table, event._guests))
+        tables.append(table)
+    if collect_stats:
+        return tables, logbook, winner, total_like_preferences, total_dislike_preferences
+    else:
+        return tables
+
+
+def guest_list_to_nested_dict(guests):
+    guests_dict = dict()
+    for guest in guests:
+        if not (guest.seating_preferences is None) and len(guest.seating_preferences) > 0:
+            pref_dict = dict()
+            for pref in guest.seating_preferences:
+                if pref.other_guest is None:
                     continue
-                pref = guest_preferences[guest_numbers[j]]
-                if not (pref is None) and pref == 0:
-                    count += 1
-        return count
+                pref_dict[pref.other_guest.number] = pref.preference.value
+        else:
+            pref_dict = None
+        guests_dict[guest.number] = pref_dict
+    return guests_dict
+
+
+def should_terminate(hall_of_fame, generation_number, total_likes, NGEN):
+    found_optimal_solution = len(hall_of_fame) > 0 and hall_of_fame[0].fitness.values == (0, total_likes)
+    return found_optimal_solution or generation_number > NGEN
+
 
-    def get_preferences_by_guest_number(self, number):
-        return self.guest_lookup[number]
+def init_population(pcls, ind_init, ind_guess, n, pct_heuristic):
+    global toolbox
+    heuristics = list([creator.Individual(table_crossover(random.choice(ind_guess), toolbox.indices())[0]) for _ in range(floor(n * pct_heuristic))])
```

Comment:
:warning: This doesn't look right.

You're calling `creator.Individual(table_crossover(random.choice(ind_guess, toolbox.indices()))`. That means:
```
To create each individual in the heuristic part of the starting population:
   Make an initial guess/heuristic of a solution
   Make a random permutation solution
   Crossover these two
   Create an individual of the resulting child
```

I would expect the initial guess to be just this:
```
To create each individual in the heuristic part of the starting population:
   Make an initial guess/heuristic of a solution
   Create an individual of the resulting heuristic
```

Reply:
I was using this to add some more randomness into the seed population, this is how I got away with only 50 in the seed